pkgs/by-name/pr/protonmail-desktop/package.nix

```diff
@@ -0,0 +1,95 @@
+{ lib, stdenv, fetchurl, wrapGAppsHook, makeWrapper, autoPatchelfHook, pkgs
+, alsa-lib
+, at-spi2-atk
+, at-spi2-core
+, atk
+, dpkg
+, glib
+, gtk3
+, libX11
+, libXtst
+, libdrm
+, mesa
+, nss
+, udev
+, xdg-utils
+, coreutils
+, xorg
+
+# command line arguments which are always set e.g "--disable-gpu"
+, commandLineArgs ? ""
+
+# Necessary for USB audio devices.
+, pulseSupport ? stdenv.isLinux
+, libpulseaudio
+
+# For GPU acceleration support on Wayland (without the lib it doesn't seem to work)
+, libGL
+
+# For video acceleration via VA-API (--enable-features=VaapiVideoDecoder,VaapiVideoEncoder)
+, libvaSupport ? stdenv.isLinux
+, libva
+}:
+
+let
+  inherit (lib) optional makeLibraryPath makeSearchPathOutput makeBinPath;
+
+  deps = [
+    alsa-lib at-spi2-atk at-spi2-core atk
+    glib gtk3 libdrm libX11 libGL
+    libXtst mesa nss udev xorg.libxcb
+  ]
+    ++ optional pulseSupport libpulseaudio
+    ++ optional libvaSupport libva;
+
+  rpath = makeLibraryPath deps + ":" + makeSearchPathOutput "lib" "lib64" deps;
+  binpath = makeBinPath deps;
+in
+stdenv.mkDerivation rec {
+  pname = "ProtonMail-desktop";
+  baseUrl = "https://proton.me/download/mail/linux/";
+  version = "beta";
+
+  src = fetchurl {
+    url = "${baseUrl}/${pname}-${version}.deb";
+    sha256 = "sha256-3FRm02Ew9GuGMXFAkwSL+FbeLyaI1SmpDU3A5eJh6TU=";
+  };
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    dpkg
+    (wrapGAppsHook.override { inherit makeWrapper; })
+  ];
+
+  buildInputs = with pkgs; [ gtk3 libnotify libdrm gdb nss mesa alsa-lib ];
+
+  unpackPhase = "dpkg-deb --fsys-tarfile $src | tar -x --no-same-permissions --no-same-owner";
+
+  installPhase = ''
+    mkdir -p $out
+    cp -av usr/* $out
+    rm -rf usr
+    runHook postInstall
+  '';
+
+  preFixup = ''
+    # Add command line args to wrapGApp.
+    gappsWrapperArgs+=(
+      --prefix LD_LIBRARY_PATH : ${rpath}
+      --prefix PATH : ${binpath}
+      --suffix PATH : ${lib.makeBinPath [ xdg-utils coreutils ]}
```

Comment:
Why did you need them? I didn't see any crashes regarding missing dependencies

Reply:
Seems those were not needed after all. makeWrapper is enough.